scripts/autogenerate-table-of-contents.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# This script is used for automatically generating any .json table of contents files
+# used for various things on the docs site, such as:
+  # - landing pages
+  # - indexing of the knowledgebase
```

Comment:
it appears to be exclusively for kb articles. can we name it as such

Reply:
Will expand this script to also generate table of contents files for the landing page. I was trying to do it in a github action but this works better.